benchmark/run/branched_filter.ex

```diff
@@ -7,17 +7,77 @@ defmodule Benchmark.Run.BranchedFilter do
   def_input_pad :input, accepted_format: _any, availability: :on_request
   def_output_pad :output, accepted_format: _any, availability: :on_request
 
-  def_options number_of_reductions: [spec: integer()]
+  def_options number_of_reductions: [spec: integer()],
+              generator: [spec: (integer() -> integer())],
+              dispatcher: [spec: (integer(), integer() -> [integer()])]
 
   @impl true
   def handle_init(_ctx, opts) do
     workload_simulation = Reductions.prepare_desired_function(opts.number_of_reductions)
-    {[], %{workload_simulation: workload_simulation}}
+
+    {[],
+     %{
+       buffers: [],
+       workload_simulation: workload_simulation,
+       generator: opts.generator,
+       dispatcher: opts.dispatcher
+     }}
   end
 
   @impl true
-  def handle_buffer(_pad, buffer, _ctx, state) do
+  def handle_buffer(_pad, buffer, ctx, state) do
+    state = %{state | buffers: state.buffers ++ [buffer]}
     state.workload_simulation.()
-    {[forward: buffer], state}
+    how_many_buffers_to_output = state.generator.(length(state.buffers))
```

Comment:
Just a code formatting suggestion
```suggestion
    how_many_buffers_to_output = length(state.buffers) |> state.generator.()
```

Reply:
done here and in the `linear_filter.ex`